### PR TITLE
Rich history UX fixes

### PR DESCRIPTION
**Pull request by jessover9000**
UX fixes for [#22729](https://github.com/grafana/grafana/pull/22729):
 
New design in Figma: https://www.figma.com/file/5ymYczshehFNPQIBI1fKpV/Rich-history-in-Explore?node-id=382%3A17
New query card design that addresses
- Delete button for each query
- Add name of datasource/icon to each query
- Make it more clear, what is going to happen when user clicks on the query card (animation/run query) – Added "Run query" button to indicate primary action
- Renamed datasource / saved query for nonexistent datasource: "Inactive"/limited option card design

Background colour of cards: 
- Background of the rich history drawer: same as page background. 
- background color for regular cards: white. 
- background color for “missing datasource” card: #E9EDF2 (gray95).

To be solved on the technical side:
- When I hide and show the query history, the filters are reset – we should save the last filters the user had
- ~~Remove queries in history that belong to removed datasources (only removed, not renamed datasources)~~ 
_edit by Ivana: Same as with renamed datasources_

We decided not to tackle:
- Data source behaviour in Settings feels like it belongs with the Query history tab
The behaviour should apply not only to the query history but also to the saved queries. We think that toggling locked/unlocked datasources for each tab separately is more confusing than just handling it globally for the entire history.

- Move save/cancel button to the right of the comment input on history panel widths >=768px to save vertical space
The button arrangement would need to flip between large and small screen sizes, so dropped this one.

### CODE REVIEW

File: public/app/core/store.ts [hunk 1]
```diff
@@ -26,6 +26,7 @@ export class Store {
         console.error(`Error parsing store object: ${key}. Returning default: ${def}. [${error}]`);
       }
     }
+
```
Comment (ivanahuckova) [inline public/app/core/store.ts]:
```suggestion
```

File: public/app/features/explore/RichHistory/RichHistory.tsx [hunk 2]
```diff
@@ -30,7 +30,7 @@ export const sortOrderOptions = [
   { label: 'Data source Z-A', value: SortOrder.DatasourceZA },
 ];
 
-interface RichHistoryProps extends Themeable {
+export interface RichHistoryProps extends Themeable {
```
Comment (ivanahuckova) [inline public/app/features/explore/RichHistory/RichHistory.tsx]:
```suggestion
interface RichHistoryProps extends Themeable {
```

**Comment by ivanahuckova** [comment]
@aocenas @Estrax @jessover9000 PR is ready for review.

**Review by torkelo — COMMENTED** [review]
Not super thrilled with yet another new card design without addressing the problem of our many existing cards with actions. We need to unify our list item designs not add more (the reason why https://github.com/grafana/grafana/pull/21121 was paused)  

But it does look a lot better :) Need to keep this card design in mind and update it as well when we get a unified card design & shared component

File: public/app/core/utils/richHistory.ts [hunk 3]
```diff
@@ -251,3 +259,31 @@ export function mapQueriesToHeadings(query: RichHistoryQuery[], sortOrder: SortO
 
   return mappedQueriesToHeadings;
 }
+
+/* Create datasource list with images. If specific datasource retrieved from Rich history is not part of
+ * exploreDatasources add generic datasource image and add property isRemoved = true.
+ */
+export function createDatasourceListWithImages(queriesDatasources: string[]) {
```
Comment (aocenas) [inline public/app/core/utils/richHistory.ts]: Just to double check, we already should have something like this for datasource picker in explore (and panel edit) where we show both name and icon so I wonder if we do not have basically the same function somewhere already.

**Comment by aocenas** [comment]
@ivanahuckova lets first merge test in https://github.com/grafana/grafana/pull/22722 so we can then add/change tests for this PR.

File: public/app/core/utils/richHistory.ts (hunk 3 above)
Comment (ivanahuckova) [reply public/app/core/utils/richHistory.ts]: Yes, you are right. This is slightly different functionality cause is creates structured data source list with required properties. I have updated the naming, so it makes more sense.

**Comment by ivanahuckova** [comment]
@aocenas
- merged master with test coverage
- added test for the new functionality

File: public/app/core/utils/richHistory.ts [hunk 4]
```diff
@@ -257,3 +265,31 @@ export function mapQueriesToHeadings(query: RichHistoryQuery[], sortOrder: SortO
 
   return mappedQueriesToHeadings;
 }
+
+/* Create datasource list with images. If specific datasource retrieved from Rich history is not part of
+ * exploreDatasources add generic datasource image and add property isRemoved = true.
+ */
+export function createDatasourcesList(queriesDatasources: string[]) {
```
Comment (aocenas) [inline public/app/core/utils/richHistory.ts]: Can we add a test for this if we are keeping it?

**Review by aocenas — APPROVED** [review]
Can be merged but would like to add the test, we can add it later though if we want to make it for the release tomorrow.

**Review by Estrax — APPROVED** [review]
Great job. Let's merge this.

**Pull request merged**